### PR TITLE
bug 1352232: Remove Vagrant references from documentation

### PR DESCRIPTION
**Pull request by jwhitlock**
This builds on and replaces PR #4213 (nitpicks and flake8), and then remove references to Vagrant from the documentation, and fix some small documentation issues:

- Update status of Dockerization
- Move subjects under Advanced Configuration up, so they appear in sidebars
- Update the docs for updating localization files to reflect my current process
- Convert performance testing instructions to Docker

### CODE REVIEW

File: docs/tests-performance.rst [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
-and the instructions have not been updated for the Docker development environment.
+These tests are not maintained or run during the current development process.
+Some of the common URLs are not included in the sample database.
 
 Running tests
 -------------
 .. note:: **DO NOT RUN LOCUST TESTS AGAINST PRODUCTION**
 
-1. Start locust from the development VM::
+1. Create a file ``docker-compose.locust.yml`` in the root folder (same folder
```
Comment (escattone) [inline docs/tests-performance.rst]: Replace leading ``1.`` with auto-numbering ``#.``

File: docs/tests-performance.rst [hunk 2]
```diff
@@ -1,25 +1,43 @@
...
+        environment:
+          - LOCUST_MODE=standalone
+          - TARGET_URL=http://web:8000
+        volumes:
+          - ./tests/performance/:/tests
+        ports:
+          - "8089:8089"
 
-2. Go to `http://developer-local.allizom.org:8089/ <http://developer-local.allizom.org:8089/>`_ UI for controlling:
+1. Edit ``.env`` to update docker-compose::
```
Comment (escattone) [inline docs/tests-performance.rst]: Replace leading ``1.`` with auto-numbering ``#.``

File: docs/tests-performance.rst [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
+
+    COMPOSE_FILE=docker-compose.yml:docker-compose.locust.yml
+
+1. Restart with ``make up``
```
Comment (escattone) [inline docs/tests-performance.rst]: Replace leading ``1.`` with auto-numbering ``#.``

File: docs/tests-performance.rst [hunk 4]
```diff
@@ -1,25 +1,43 @@
...
+
+1. Load the Locust UI at http://localhost:8089, and select:
```
Comment (escattone) [inline docs/tests-performance.rst]: Replace leading ``1.`` with auto-numbering ``#.``

File: docs/development.rst [hunk 5]
```diff
@@ -443,7 +353,7 @@ them against your local Docker instance, first do the following:
 #. Load the latest sample database (see :ref:`provision-the-database`).
 #. Ensure that the test document "en-US/docs/User:anonymous:uitest" has been
    rendered (all of its macros have been executed). You can check this by
-   browsing to `http://localhost:8000/en-US/docs/User:anonymous:uitest`_. If
```
Comment (escattone) [inline docs/development.rst]: I think I was the cause of this. Thanks for the clean-up.

File: docs/installation.rst [hunk 6]
```diff
@@ -1,30 +1,37 @@
...
+* The Docker development environment doesn't fully support a 'production-like'
+  environment. For example, we don't have a documented configuration for
+  running with an SSL connection.
+* When the master branch is updated, the ``kuma_base`` image is refreshed and
+  published to `quay.io`_. This image contains system packages and
+  third-party libraries.
+* Our TravisCI_ builds include a target that build Docker containers and runs
+  the tests inside.
+* Our Jenkins_ server builds and publishes Docker images, and runs integration
+  tests using Docker.
```
Comment (escattone) [inline docs/installation.rst]: Nice additions 👍

File: docs/tests-performance.rst [hunk 7]
```diff
@@ -1,25 +1,43 @@
...
+        build: ./tests/performance
+        depends_on:
+          - web
+        environment:
+          - LOCUST_MODE=standalone
+          - TARGET_URL=http://web:8000
+        volumes:
+          - ./tests/performance/:/tests
+        ports:
+          - "8089:8089"
```
Comment (escattone) [inline docs/tests-performance.rst]: Nice addition 👍

File: docs/development.rst [hunk 8]
```diff
@@ -334,21 +246,21 @@ To connect to the database specified in ``DATABASE_URL``, use::
...
+Generating Production Assets
+============================
 Kuma will automatically run in debug mode, with the ``DEBUG`` setting turned to
 ``True``. Setting ``DEBUG=False`` will put you in production mode and
 generate/use minified (compressed) and versioned (hashed) assets. To
 emulate production, and test compressed and hashed assets locally:
 
 #. Set the environment variable ``DEBUG=false``.
-#. Start (``docker-compose up -d``) or restart (``docker-compose restart`)
+#. Start (``docker-compose up -d``) or restart (``docker-compose restart``)
```
Comment (escattone) [inline docs/development.rst]: Good catch.

**Comment by jwhitlock** [comment]
New commit to update ``docs/tests-performance.rst`` to use auto-numbered lists, and to correctly indent the bullets in the last step.

**Pull request merged**